lib/tbot/config/cli.go

```diff
@@ -0,0 +1,579 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package config
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+	"github.com/gravitational/trace"
+)
+
+type globalArgs struct {
+	FIPS bool
+
+	// These properties are not applied to the config.
+
+	ConfigPath string
+	Debug      bool
+}
+
+func (g *globalArgs) ApplyConfig(cfg *BotConfig) error {
+	if g.FIPS {
+		cfg.FIPS = g.FIPS
+	}
+
+	if g.Debug {
+		cfg.Debug = g.Debug
+	}
+
+	return nil
+}
+
+// sharedStartArgs are arguments that are shared between all modern `start` and
+// `configure` subcommands.
+type sharedStartArgs struct {
+	ProxyServer     string
+	JoinMethod      string
+	Insecure        bool
+	Token           string
+	CAPins          []string
+	CertificateTTL  time.Duration
+	RenewalInterval time.Duration
+	Storage         string
+
+	LogFormat string
+	Oneshot   bool
+	DiagAddr  string
+}
+
+// newSharedStartArgs initializes shared arguments on the given parent command.
+func newSharedStartArgs(cmd *kingpin.CmdClause) *sharedStartArgs {
+	args := &sharedStartArgs{}
+
+	joinMethodList := fmt.Sprintf(
+		"(%s)",
+		strings.Join(SupportedJoinMethods, ", "),
+	)
+
+	cmd.Flag("proxy-server", "Address of the Teleport Proxy Server.").Envar(proxyServerEnvVar).StringVar(&args.ProxyServer)
```

Comment:
It occurs to me --auth-server is not actually deprecated, just no longer preferred, and will need to be supported here. Will re-add.